src/clj/worker/listener.clj

```diff
@@ -12,12 +12,20 @@
             [mount.core :as mount :refer [defstate]]
             [shared.utils :refer [file-exists?]]
             [taoensso.timbre :as log])
-  (:import [com.spread.parsers BayesFactorParser ContinuousTreeParser DiscreteTreeParser TimeSlicerParser]))
+  (:import [com.spread.parsers BayesFactorParser ContinuousTreeParser DiscreteTreeParser TimeSlicerParser]
+           [com.spread.progress IProgressObserver IProgressReporter]))
 
 (declare listener)
 
 (defonce tmp-dir "/tmp")
 
+(defn new-progress-handler [callback]
+  (proxy [IProgressObserver] []
+    (init [^IProgressReporter reporter]
+      (.registerProgressObserver reporter this))
```

Comment:
I'm not sure I understand the init interface method here, it is like the observer is registering itself on some observable.
Normally something else is registering observers into observables.
I thought the idea of the init was a method that observables/reporters call from observers to set up initial state before starting to call handleProgress.

Reply:
Done